packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
```

Comment:
I really don't like hiding actual variant values with this shared types. Can't we just list all options as normal union?

Reply:
Personally I like the certainty we get from extending the shared type, that all variants across components are the same and that we don't have to update them in multiple places. The `Simplify` type helps write out all the variants in your editor.

Reply:
I don't think we'll have the same list of variants for all components, they will always differ because they're contextual and I'd personally prefer to have the whole list of all options in the corresponding component file.

+ I'm not sure if <Props /> table and Storybook table will be able to parse this and show all possible options to the end user.

Reply:
It works with both of them without issues.